test/test_torch.py

```diff
@@ -17272,6 +17272,60 @@ def test_reciprocal_complex_extremal(self, device, dtype):
 
         self.compare_with_numpy(torch.reciprocal, np.reciprocal, vals, device, dtype)
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(torch.bool, *(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes()))
+    def test_copysign(self, device, dtype):
+        def _test_copysign_numpy(a, b):
+            torch_result = torch.copysign(a, b)
+
+            if a.dtype == torch.bfloat16:
+                np_a = a.to(torch.float).cpu().numpy()
+            else:
+                np_a = a.cpu().numpy()
+
+            if b.dtype == torch.bfloat16:
+                np_b = b.to(torch.float).cpu().numpy()
+            else:
+                np_b = b.cpu().numpy()
+            expected = torch.from_numpy(np.copysign(np_a, np_b))
+
+            # To handle inconsistencies of type promotion between PyTorch and Numpy
+            # Applied for both arguments having integral precision and bfloat16
+            types = [torch.bool, torch.bfloat16] + torch.testing.get_all_int_dtypes()
+            if a.dtype in types or b.dtype in types:
+                promoted_type = torch.promote_types(torch_result.dtype, expected.dtype)
+                torch_result = torch_result.to(promoted_type)
```

Comment:
No need to cast the `torch_result` - it should already be the correct type, right?

Reply:
I would say cast the numpy result may not useful since Numpy generate higher precision than torch. https://github.com/pytorch/pytorch/pull/46396#issuecomment-715442939

Reply:
Oh, I see what is going on here. That makes sense

Reply:
> I would say cast the numpy result may not useful since Numpy generate higher precision than torch. [#46396 (comment)](https://github.com/pytorch/pytorch/pull/46396#issuecomment-715442939)

I am sorry if it's confusing. We do have some edge cases that torch has higher precision like bool and bool. torch promotes to float32 but numpy promotes to float16. So, I think we still need to keep both casting.

Originally we want to use two inputs to infer the promoted type, but it's not working for two integral types. So I use the the dtypes from results to infer the promoted type.

Reply:
This makes sense to me, thanks for the explanation -- let's keep it as-is